src/unit/test_fifo.c

```diff
@@ -0,0 +1,261 @@
+#include "../fifo.c"
```

Comment:
According to Maddy, I don't think this should be necessary.  Code should all be build together.  Just include `fifo.h`